Clearer error message and clean up for Product Categories

EdwardLi-coder opened the pull request:
#### What? Why?

- Closes #12773

I found that the code had no place to use spree_classification_primary_taxon_error and invalid_taxonomy_id, so I deleted them


#### What should we test?
<!-- List which features should be tested and how.
     This can be similar to the Steps to Reproduce in the issue.
     Also think of other parts of the app which could be affected
     by your change. -->

As described in the issue:

* Admin setting page should work properly with new taxons options
* All crud actions should be available in admin settings for taxons

#### Release notes

<!-- Please select one for your PR and delete the other. -->

Changelog Category (reviewers may add a label for the release notes):

- [x] User facing changes
- [ ] API changes (V0, V1, DFC or Webhook)
- [ ] Technical changes only
- [ ] Feature toggled

<!-- Choose a pull request title above which explains your change to a
     a user of the Open Food Network app. -->

The title of the pull request will be included in the release notes.


#### Dependencies
<!-- Does this PR depend on another one?
     Add the link or remove this section. -->



#### Documentation updates
<!-- Are there any wiki pages that need updating after merging this PR?
     List them here or remove this section. -->

mkllnk commented on config/locales/ar.yml:
```diff
@@ -2653,7 +2653,6 @@ ar:
   spree_admin_single_enterprise_hint: "لمحة: للسماح للأشخاص بالعثور عليك ، قم بتشغيل الرؤية الخاصة بك أسفل"
   spree_admin_eg_pickup_from_school: "على سبيل المثال. &quot;البيك اب من المدرسة الابتدائية&quot;"
   spree_admin_eg_collect_your_order: "على سبيل المثال. &quot;يرجى جمع طلبك من 123 Imaginary St، Northcote ، 3070&quot;"
-  spree_classification_primary_taxon_error: "Taxon %{taxon} هو التصنيف الأساسي %{product} ولا يمكن حذفه"
```
We never change other locales than en.yml because they are updated automatically through our Transifex integration.

EdwardLi-coder replied:
> We never change other locales than en.yml because they are updated automatically through our Transifex integration.

OK.I will update it.

mkllnk reviewed (changes requested):
Thank you for tackling this. Could you reduce your change to only en.yml? The other files will get updated automatically in our release process. It will be easier to see your actual changes then as well.

mkllnk reviewed (approved):
Great, thank you!

Note that we usually don't merge master but rebase on master if needed. You don't have to do that for this pull request though.

chahmedejaz reviewed (changes requested):
Looks good, thanks. However as per the issue, can you please confirm if `spree.new_taxon` translation is being used?

EdwardLi-coder commented:
> Great, thank you!
> 
> Note that we usually don't merge master but rebase on master if needed. You don't have to do that for this pull request though.

Ok,I see.Thank you tell me.

EdwardLi-coder commented:
> Looks good, thanks. However as per the issue, can you please also confirm if `spree.new_taxon` translation is being used?

Maybe this use it.I'm not sure.
` %li
    = button_link_to t(".new_taxon"), spree.new_admin_taxon_url, icon: 'icon-plus', id: 'admin_new_taxon_link'`

chahmedejaz commented:
> %li = button_link_to t(".new_taxon"), spree.new_admin_taxon_url, icon: 'icon-plus', id: 'admin_new_taxon_link'

I think the following translation is being used here because it's using Rail's lazy translation lookup and it should get translated to this key based upon the file path in which it's used, `spree.admin.taxons.index.new_taxon`:
https://github.com/openfoodfoundation/openfoodnetwork/blob/917079931e942d57f97ba9c7af939d3b6065ae8c/config/locales/en.yml#L4587

I think we should validate this and remove `spree.new_taxon` as well if it's unused. Would be good to please do that? Thanks.

EdwardLi-coder commented:
> > %li = button_link_to t(".new_taxon"), spree.new_admin_taxon_url, icon: 'icon-plus', id: 'admin_new_taxon_link'
> 
> I think the following translation is being used here because it's using [Rail's lazy translation lookup](https://stackoverflow.com/questions/34263083/i18n-rails-lazy-look-up-from-a-helper) and it should get translated to this key based upon the file path in which it's used, `spree.admin.taxons.index.new_taxon`:
> 
> https://github.com/openfoodfoundation/openfoodnetwork/blob/917079931e942d57f97ba9c7af939d3b6065ae8c/config/locales/en.yml#L4587
> 
> I think we should validate this and remove `spree.new_taxon` as well if it's unused. Would be good to please do that? Thanks.

I have already deleted spree.new_taxon.Thanks

chahmedejaz commented:
> > > %li = button_link_to t(".new_taxon"), spree.new_admin_taxon_url, icon: 'icon-plus', id: 'admin_new_taxon_link'
> > 
> > I think the following translation is being used here because it's using [Rail's lazy translation lookup](https://stackoverflow.com/questions/34263083/i18n-rails-lazy-look-up-from-a-helper) and it should get translated to this key based upon the file path in which it's used, `spree.admin.taxons.index.new_taxon`:
> > 
> > https://github.com/openfoodfoundation/openfoodnetwork/blob/917079931e942d57f97ba9c7af939d3b6065ae8c/config/locales/en.yml#L4587
> > 
> > I think we should validate this and remove `spree.new_taxon` as well if it's unused. Would be good to please do that? Thanks.
> 
> I have already deleted spree.new_taxon.Thanks

Really sorry, but I believe you have pushed that change in another branch which might not be relevant to that change.
https://github.com/EdwardLi-coder/openfoodnetwork/tree/change_colour_of_complete_order

Can you please validate and push that change in this branch so that the relevant issue's changes are within their respective branches?

EdwardLi-coder commented:
> > > > %li = button_link_to t(".new_taxon"), spree.new_admin_taxon_url, icon: 'icon-plus', id: 'admin_new_taxon_link'
> > > 
> > > 
> > > I think the following translation is being used here because it's using [Rail's lazy translation lookup](https://stackoverflow.com/questions/34263083/i18n-rails-lazy-look-up-from-a-helper) and it should get translated to this key based upon the file path in which it's used, `spree.admin.taxons.index.new_taxon`:
> > > https://github.com/openfoodfoundation/openfoodnetwork/blob/917079931e942d57f97ba9c7af939d3b6065ae8c/config/locales/en.yml#L4587
> > > 
> > > I think we should validate this and remove `spree.new_taxon` as well if it's unused. Would be good to please do that? Thanks.
> > 
> > 
> > I have already deleted spree.new_taxon.Thanks
> 
> Really sorry, but I believe you have pushed that change in another branch which might not be relevant to that change. https://github.com/EdwardLi-coder/openfoodnetwork/tree/change_colour_of_complete_order
> 
> Can you please validate and push that change in this branch so that the relevant issue's changes are within their respective branches?

Thank you for bringing this to my attention. I apologize for the confusion earlier. I have now verified and corrected the issue.

chahmedejaz commented:
No worries @EdwardLi-coder, all good now. Thanks♥️

drummer83 commented:
Hi @EdwardLi-coder,
Thank you for finishing this up! :muscle: 

I have tested the product categories and couldn't find any problems.

Also the new error message is shown:
![image](https://github.com/user-attachments/assets/3ff61b12-efba-4863-a9a5-491d67e9ad68)

Excellent! Merging! :tada: 
Thanks again!
